backend/controllers/users.js

```diff
@@ -176,16 +179,69 @@ usersRouter.post(
       });
     }
 
-    //Check user is exist or not. If not, return with error message.
+  //Check user is exist or not. If not, return with error message.
+  // Update user register API
+  const registerUser = async (request, response) => {
+  try {
+    // Check if the user already exists
     const userInfo = request.body;
     const userExists = await User.findOne({ $or: [{ email: userInfo.email }, { username: userInfo.username }] });
-    if (!userExists) {
+
+    if (userExists) {
       return response.status(400).json({
         status: "Fail",
-        error: "User doesn't exist",
+        error: "User already exists",
```

Comment:
If user is exist then we should check UserApp.
So we will get appId from API body, so we should check relationship between user and app is exist or not.
If relationship is exist then return user is exist.
But if relationship is not exist then should update user Activity, userApp and UserRank model.